masonry/src/core/widget_mut.rs

```diff
@@ -51,31 +51,38 @@ impl<W: Widget + ?Sized> WidgetMut<'_, W> {
         }
     }
 
-    /// Returns `true` if the widget has a property of type `T`.
-    pub fn get_prop<T: Property>(&self) -> Option<&T> {
-        self.ctx.properties.get::<T>()
-    }
-
-    /// Get value of property `T`, or `None` if the widget has no `T` property.
+    /// Returns `true` if the widget has a local property of type `T`.
+    ///
+    /// Does not check default properties.
     pub fn contains_prop<T: Property>(&self) -> bool {
         self.ctx.properties.contains::<T>()
     }
 
-    /// Get value of property `T`, or `None` if the widget has no `T` property.
-    pub fn get_prop_mut<T: Property>(&mut self) -> Option<&mut T> {
-        self.widget
-            .property_changed(&mut self.ctx.update_mut(), TypeId::of::<T>());
-        self.ctx.properties.get_mut::<T>()
+    /// Get value of property `T`.
+    ///
+    /// If the widget has an entry for `P`, returns that entry.
+    /// If the default property set has an entry for `P`, returns that entry.
+    /// Otherwise returns [`Property::static_default()`].
+    pub fn get_prop<T: Property>(&self) -> &T {
+        self.ctx.properties.get::<T>()
     }
 
     /// Set property `T` to given value. Returns the previous value if `T` was already set.
+    ///
+    /// Does not affect default properties.
+    ///
+    /// This also calls [`Widget::property_changed`] with the matching type id.
```

Comment:
It is confusing that this calls the notification prior to the value being set ...

Reply:
XAML's `DependencyObject` has a `DependencyPropertyChangedEventArgs` which gets the old and new values.

Reply:
Wouldn't help. Some things need the old value, some need the new. This is a future concern though.